server/src/main/java/org/candlepin/controller/ManifestManager.java

```diff
@@ -0,0 +1,388 @@
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.controller;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.io.FileUtils;
+import org.candlepin.audit.EventFactory;
+import org.candlepin.audit.EventSink;
+import org.candlepin.common.exceptions.BadRequestException;
+import org.candlepin.common.exceptions.IseException;
+import org.candlepin.common.exceptions.NotFoundException;
+import org.candlepin.guice.PrincipalProvider;
+import org.candlepin.model.Consumer;
+import org.candlepin.model.ConsumerCurator;
+import org.candlepin.model.EntitlementCurator;
+import org.candlepin.model.ImportRecord;
+import org.candlepin.model.ManifestFileRecordType;
+import org.candlepin.pinsetter.tasks.ExportJob;
+import org.candlepin.pinsetter.tasks.ImportJob;
+import org.candlepin.model.Owner;
+import org.candlepin.sync.ConflictOverrides;
+import org.candlepin.sync.ExportCreationException;
+import org.candlepin.sync.ExportResult;
+import org.candlepin.sync.Exporter;
+import org.candlepin.sync.Importer;
+import org.candlepin.sync.ImporterException;
+import org.candlepin.sync.ManifestFileService;
+import org.candlepin.sync.ManifestServiceException;
+import org.candlepin.sync.file.ManifestFile;
+import org.candlepin.util.Util;
+import org.quartz.JobDetail;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xnap.commons.i18n.I18n;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.google.inject.persist.Transactional;
+
+/**
+ * This class serves as a controller layer for manifest export and import and encapsulates
+ * the {@link Importer} and {@link Exporter} functionality so that it isn't spread out
+ * across multiple classes.
+ *
+ */
+public class ManifestManager {
+
+    private static Logger log = LoggerFactory.getLogger(ManifestManager.class);
+    private ManifestFileService manifestFileService;
+    private Exporter exporter;
+    private Importer importer;
+    private EntitlementCurator entitlementCurator;
+    private PoolManager poolManager;
+    private ConsumerCurator consumerCurator;
+    private PrincipalProvider principalProvider;
+    private I18n i18n;
+    private EventSink sink;
+    private EventFactory eventFactory;
+
+    @Inject
+    public ManifestManager(ManifestFileService manifestFileService, Exporter exporter, Importer importer,
+        ConsumerCurator consumerCurator, EntitlementCurator entitlementCurator, PoolManager poolManager,
+        PrincipalProvider principalProvider, Provider<I18n> i18nProvider, EventSink eventSink,
```

Comment:
You don't need to inject `I18n` as a provider unless this class sits in a larger scope that request-scope.